Sources/libcggen/BCCGGenerator.swift

```diff
@@ -0,0 +1,341 @@
+import BCCommon
+import Foundation
+
+private protocol ByteCodable {
+  var byteCode: [UInt8] { get }
+}
+
+extension UInt32: ByteCodable {
+  var byteCode: [UInt8] {
+    var littleEndian = self.littleEndian
+    let count = MemoryLayout<Self>.size
+    let bytePtr = withUnsafePointer(to: &littleEndian) { pointer in
+      pointer.withMemoryRebound(to: UInt8.self, capacity: count) { pt in
+        UnsafeBufferPointer(start: pt, count: count)
```

Comment:
unsafe pointers should not escape `withUnsafePointer` body, that's an undefined behaviour. Actually, this code can be much simpler:
```
var byteCode: [UInt8] {
  withUnsafeBytes(of: littleEndian, Array.init)
}
```